contracts/Market.sol

```diff
@@ -182,7 +185,7 @@ contract Market is PausableAccessPreset {
     uint256[] memory,
     bytes memory
   ) external returns (bytes4) {
-    // todo revert if Market.onERC1155BatchReceived sender is not the removal contract
+    require(_msgSender() == address(_removal), "Sender not Removal contract");
```

Comment:
we use revert strings here instead of custom errors because the inherited code that catches errors in this `onERC1155(Batch)Received` hook can't match custom errors, and we end up getting the misleading error "Transfer to Non-ERC1155 Receiver".